include/aspect/heating_model/interface.h

```diff
@@ -126,67 +164,132 @@ namespace aspect
+         * let these objects read their parameters as well.
+         */
+        void
+        parse_parameters (ParameterHandler &prm);
+
+        /**
+         * A function that is called at the beginning of each time step,
+         * calling the update function of the individual heating models.
+         */
+        void
+        update ();
+
+
+        /**
+         * A function that calls the evaluate function of all the individual
+         * heating models.
+         */
+        void
+        execute (const typename aspect::MaterialModel::Interface<dim>::MaterialModelInputs &material_model_inputs,
+                 const typename aspect::MaterialModel::Interface<dim>::MaterialModelOutputs &material_model_outputs,
+                 HeatingModel::HeatingModelOutputs &heating_model_outputs) const;
+
+
+        /**
+         * A function that is used to register heating model objects in such
+         * a way that the Manager can deal with all of them without having to
+         * know them by name. This allows the files in which individual
+         * plugins are implemented to register these plugins, rather than also
+         * having to modify the Manager class by adding the new mesh
+         * refinement class.
+         *
+         * @param name A string that identifies the heating model
+         * @param description A text description of what this model does and that
+         * will be listed in the documentation of the parameter file.
+         * @param declare_parameters_function A pointer to a function that can be
+         * used to declare the parameters that this heating model wants to read
+         * from input files.
+         * @param factory_function A pointer to a function that can create an
+         * object of this heating model.
+         *
+         * @ingroup HeatingModels
+         */
+        static
+        void
+        register_heating_model (const std::string &name,
+                                const std::string &description,
+                                void (*declare_parameters_function) (ParameterHandler &),
+                                Interface<dim> *(*factory_function) ());
 
+        /**
+         * Return a list of names of all implemented heating models, separated by
+         * '|' so that it can be used in an object of type Patterns::Selection.
+         */
+        std::string
+        get_all_heating_model_names () const;
+
+        /**
+         * Return a list of names of all heating models currently used in the
+         * computation, as specified in the input file.
+         */
+        std::vector<std::string>
+        get_active_heating_model_names () const;
+
+        /**
+         * Return a list of pointers to all heating models currently used in the
+         * computation, as specified in the input file.
+         */
+        std::list<std_cxx11::shared_ptr<Interface<dim> > >
+        get_heating_models () const;
```

Comment:
If you call the previous function `get_*active*_heating_model_names` then maybe this function should also be called `get_active_heating_models`.